TrappingRainWater/main.h

```diff
@@ -0,0 +1,11 @@
+#ifndef MAIN_H
+#define MAIN_H
+
+#include <vector>
+
+class Solution {
+ public:
+  int trap(std::vector<int>& height);
```

Comment:
does height change? 
If not, make it const &
Better yet, use std::span<const int>.